### ISSUE TITLE
Some loops in watershed delineation could use timed logging

### ISSUE THREAD

**Issue by phargogh** [natcap/pygeoprocessing#40]
If I'm delineating a large number of geometries on a large landscape, there are a few stretches of watershed delineation that have no logging whatsoever, and yet it's clear from CPU and disk usage that something is happening.  It would be nice to add some progress logging to this.

To reproduce, run the InVEST 3.8.0 DelineateIt on the Colombia DEM with the Colombia municipalities vector.

**Comment by richpsharp** [comment]
I'm assigning you @phargogh and reclassifying as an enhancement since it's not really something incorrect about the code base.

**Comment by phargogh** [comment]
Sounds good!  I had added the `bug` label because that seemed like the clearest way to indicate that it should be merged into `master`.  Since I added this issue, we instead seem to have settled on the use of milestones to organize which issues should be merged into which branches, which makes more sense anyways.

**Comment by richpsharp** [comment]
Hi @phargogh, I looked into this and added a timed logger on the outer loop that processes the feature count, then saw a little bit down you'd already done the same thing. The other loop I saw was one merging the geometry together which I wouldn't think would take that long, plus there's also the polygonalize call which I think should be covered by the per-feature logging. I thought this would be obvious, but now I'm not sure what will be useful here. I'm passing back to you since I think you have some insight into what's useful here.
